test/e2e/tree-shaking-flow/fixture/src/main.js

```diff
@@ -0,0 +1,59 @@
+// @flow
+
+/* Includes type imports from both node_modules and userland but as
+ * standalone type imports and mixed imports:
+ *    import type {AType, BType} from ...
+ *    import {type AType, type BType, C, D} from ...
+ */
+
+// Import only types from dependency
+import type {
+  UnusedDependencyType,
+  UsedDependencyType,
+} from 'fixture-pkg';
+
+// Import mixed types and values from dependency
+import {
+  type UnusedDependencyOtherType,
+  type UsedDependencyOtherType,
+  unusedDependencyValue,
+  usedDependencyValue
+} from 'fixture-pkg';
+
+// Import only types from userland
+import type {
+  UnusedUserlandType,
+  UsedUserlandType
+} from './user-code.js';
+
+// Import mixed types and values from userland
+import {
+  type UnusedUserlandOtherType,
+  type UsedUserlandOtherType,
+  unusedUserlandValue,
+  usedUserlandValue
+} from './user-code.js';
+
+if (__NODE__) {
+  // unused in the browser
+  //  - node modules
+  const unusedDepVal: UnusedDependencyType = { shouldBeUnused: unusedDependencyValue };
+  const unusedDepVal2: UnusedDependencyOtherType = { alsoShouldBeUnused: unusedDependencyValue };
+  //  - userland
+  const unustedUserlandVal: UnusedUserlandType = { shouldBeUnused: unusedUserlandValue };
+  const unustedUserlandVal2: UnusedUserlandOtherType = { alsoShouldBeUnused: unusedUserlandValue };
+
+  console.log(unusedDepVal, unusedDepVal2, unustedUserlandVal, unustedUserlandVal2);
+}
+
+// used in the browser
+//  - node modules
+const usedDepVal: UsedDependencyType = { shouldBeUsed: usedDependencyValue};
```

Comment:
Lines `51-52` here cause the vendor bundle to leak the unused import from `fixture-pckg` (`__FIXTURE_DEPENDENCY_UNUSED__` which should be used only in the server bundle):

```js
/*! no static exports found */function(e,o,s){"use strict";Object.defineProperty(o,"__esModule",{value:!0});o.usedDependencyValue="__FIXTURE_DEPENDENCY_USED__",o.unusedDependencyValue="__FIXTURE_DEPENDENCY_UNUSED__"}}]);
```

Removing these two lines tree shakes the import for `fixture-pckg`.